demos/radio.html

```diff
@@ -0,0 +1,84 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta charset="utf-8">
+    <title>MDL Radio Demo</title>
+    <script src="assets/material-design-lite.css.js" charset="utf-8"></script>
+    <style type="text/css">
+    .surface {
+      box-shadow: 0px 2px 1px -1px rgba(0, 0, 0, 0.2),
+      0px 1px 1px 0px rgba(0, 0, 0, 0.14),
+      0px 1px 3px 0px rgba(0, 0, 0, 0.12);
+
+      position: relative;
+    }
+     section {
+       margin: 48px;
+     }
+     body {
+       font-family: sans-serif;
+     }
+     button {
+       display: block;
+       margin: 16px 0;
+     }
+     .mdl-radio {
+       margin: 8px;
+     }
+     .mdl-radio span {
+       display: inline-block;
+       line-height: 24px;
+       margin-left: 4px;
+       vertical-align: top;
+     }
+    </style>
+  </head>
+  <body>
+    <main>
+      <section>
+        <label for="radio1" class="mdl-radio">
```

Comment:
The main advantage is getting the connection between the label and the input (important for screen readers) without having to use the `for` property (we were doing both in v1, but that's completely unnecessary). That way you don't need to attach IDs to everything.

In addition, it works as a very useful root element, so that you don't need to wrap the `input` and the `label` in a `div` or something.

I'd still like to get in touch with our accessibility expert regarding this, though.